dll/win32/ws2_32/src/addrconv.c

```diff
@@ -492,3 +492,135 @@ WSANtohs(IN SOCKET s,
     return SOCKET_ERROR;
 }
 
+INT
+WSAAPI
+InetPtonW(_In_ INT Family,
+    _In_ PCWSTR pszAddrString,
+    _Out_writes_bytes_(sizeof(IN_ADDR6)) PVOID pAddrBuf)
+{
+    NTSTATUS Status;
+    PCWSTR ch;
+
+    if (!pszAddrString || !pAddrBuf)
+    {
+        SetLastError(WSAEFAULT);
+        return -1;
+    }
+
+    switch (Family)
+    {
+        case AF_INET:
+            Status = RtlIpv4StringToAddressW(pszAddrString, TRUE, &ch, pAddrBuf);
+            break;
+        case AF_INET6:
+            Status = RtlIpv6StringToAddressW(pszAddrString, &ch, pAddrBuf);
+            break;
+        default:
+            SetLastError(WSAEAFNOSUPPORT);
+            return -1;
+    }
+
+    if (!NT_SUCCESS(Status) || (*ch != 0)) {
```

Comment:
```suggestion
    if (!NT_SUCCESS(Status) || (*ch != 0))
    {
```
also you could change `(*ch != 0)` to `!*ch`.

Reply:
`(*ch != 0)` is simpler to understand than `!*ch`